misc-scripts/gifts_import/populate_protein_feature.pl

```diff
@@ -0,0 +1,350 @@
+
+my $create_index = 0;
+my $dryrun = 0;
+#my $limit = 10;
+my $limit = 0;
+
+
+GetOptions(
+  "gfhost=s"    => \$gfhost,
+  "gfport=i"       => \$gfport,
+  "gfuser=s"    => \$gfuser,
+  "gfpass=s"    => \$gfpass,
+  "gfdb=s"   => \$gfdb,
+
+  "pfhost=s"      => \$pfhost,
+  "pfport=i"      => \$pfport,
+  "pfuser=s"      => \$pfuser,
+  "pfpass=s"        => \$pfpass,
+  "pfdb=s"   => \$pfdb,
+
+  "dryrun!"         => \$dryrun,
+  "help", \&usage,
+
+);
+
+usage() if ( !defined $gfhost || !defined $gfport || !defined $gfuser || !defined $gfdb );
+usage() if ( !defined $pfhost  || !defined $pfport || !defined $pfuser || !defined $pfpass );
+
+
+my ( $giftsDB, $protfeatureDB ) = init_db();
+die unless defined($giftsDB) and defined($protfeatureDB);
+
+print "giftsDB credentials:\n";
+print Dumper($giftsDB);
+
+print "protfeatureDB credentials:\n";
+print Dumper($protfeatureDB);
+
+
+my $registry = get_registry();
+my $translation_adaptor = $registry->get_adaptor( 'human', 'core', 'translation' );
+
+
+sub init_db {
+
+# This is where the gifts info will be read from
+  my $giftsDB = {
+    user => $gfuser,
+    pass => $gfpass,
+    host => $gfhost,
+    port => $gfport,
+    dbname => $gfdb
+
+  };
+
+# This is where the gifts info will be written to
+  my $protfeatureDB = {
+    user   => $pfuser,
+    pass   => $pfpass,
+    host   => $pfhost,
+    port   => $pfport,
+    dbname => $pfdb
+  };
+
+  return ( $giftsDB, $protfeatureDB );
+}
+
+
+populate_protein_feature_db();
+
+sub get_analysis_id {
+  my ($logic_name) = @_;
+  my $analysis_id;
+  
+  my $dba = get_dba($protfeatureDB);
+  my $analysis_dba = $dba->get_AnalysisAdaptor();
+  my $analysis = $analysis_dba->fetch_by_logic_name($logic_name);
+  
+  $analysis_id = $analysis->dbID() if defined $analysis;
+  return $analysis_id;
+}
+
+
+sub populate_protein_feature_db {
+
+  my $gifts_dbh = db_connect($giftsDB );
+  my $protfeature_dbh = db_connect($protfeatureDB);
+  
+  if($gifts_dbh){
+    print "Got connection to GiftsDB \n";
+  }
+  
+  if($protfeature_dbh){
+    print "Got connection to ProtfeatureDB \n";
+  }
+  
+  my $insert_container = build_insert_sql($gifts_dbh, $protfeature_dbh);
+
+  
+  my $rows_inserted = 0;
+  my @failed_stmts = [];
+  unless($dryrun){
+
+  foreach my $insert_stmt (@$insert_container) {
+    eval{
+    my $affected = $protfeature_dbh->do($insert_stmt);
+    $rows_inserted += $affected;
+    };
+    if($@){
+      print "$@\n";
+      push(@failed_stmts, $@);
+    }
+    
+  }
+  print Dumper(\@failed_stmts);
+  print "INSERT END\n";
+  }
+
+  print "Number of rows inserted into protein_feature $protfeatureDB->{'dbname'}.protein_feature ", $rows_inserted, "\n";
+  return $rows_inserted;
+  
+  $gifts_dbh->disconnect();
+  $protfeature_dbh->disconnect();
+  
+}
+
+sub annotate_ref{
+  my $ref = shift;
+  my $ensp_id = $ref->[0];
+  my $ensp = $translation_adaptor->fetch_by_stable_id( $ensp_id ) ;
+  if($ensp){
+  $ref->[0] = $ensp->dbID;
+  $ref->[1] = 1;
```

Comment:
The discussion you're hearing behind you, we're not sure if this is a good idea, that object coming back from DBI really isn't yours to mess with. It's likely alright if it's not crashing on you so far, but it just feels a little wrong since this whole fetchrow_arrayref lays somewhere between the perl and XS C layer of the driver.

In addition, could we have a bit of inline documentation for the future, why are you just putting '1' here. Why in the next line is is length()-1.